test/lib/sstable_utils.hh

```diff
@@ -230,7 +230,7 @@ public:
     void rewrite_toc_without_scylla_component() {
         _sst->_recognized_components.erase(component_type::Scylla);
         remove_file(_sst->filename(component_type::TOC)).get();
-        _sst->write_toc(default_priority_class());
+        _sst->_storage.open(*_sst, default_priority_class());
```

Comment:
Since the plan is to implement "ownership table" (whatever it means) it's better if the temporary-TOC hack doesn't sneak into object-storage backend. So the

- open (create temporary toc)
- ...
- seal (rename temporary toc into toc)

is translated into

- open (insert into ownership state = creating)
- ...
- seal (update ownership state = sealed)

or alike

Reply:
Yes

Reply:
@xemul do we need a TOC object at all with an ownership table?
The latter can contain the set of components using a collection or even just a string encoding the set.

Reply:
> @xemul do we need a TOC object at all with an ownership table?

TOC is the list of components. In "ownership table" there will be an "entry" containing components, so I think that there will be no need in dedicated TOC component

Reply:
I think it's still worthwhile to keep it, so the sstable can be accessed by tools (or just copied into an unshared-storage scylla)